### PR TITLE
Add a mention of our nightly builds to our `README.md`.

### PR DESCRIPTION
**Pull request by chandlerc**
This provides an example of downloading and using the nightly buildings of the toolchain. I've tried to provide appropriate caveats about the fact that this is very early and not something that's really reliable. But I wanted folks to know how they can play with the nightly releases if they're interested.

I've also focused the source build instructions on the toolchain where we'd be interested in contributions, and the first paragraph on trying out Carbon in the browser with CE.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
...
-can download and try out locally, you'll instead need to build any tools
-yourself from source. We expect to have packaged releases you can try out when
-we reach our
+Carbon is an early, experimental project, and so we only have very experimental
+nightly releases of the Carbon toolchain available to download, and only on
+limited platforms. If you are using a recent Ubuntu Linux or similar (Debian,
+WSL, etc.), you can try these out by going to our
+[releases](https://github.com/carbon-language/carbon-lang/releases) page and
+download the latest nightly toolchain tar file:
+`carbon_toolchain-v0.0.0-0.nightly.YYYY.MM.DD.tar.gz`. Then you can try it out:
```
Comment (jonmeow) [inline README.md]:
```suggestion
`carbon_toolchain-0.0.0-0.nightly.YYYY.MM.DD.tar.gz`. Then you can try it out:
```

File: README.md [hunk 2]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
...
+
+```shell
+# A variable with the specific nightly version:
+VERSION="v0.0.0-0.nightly.YYYY.MM.DD"
```
Comment (jonmeow) [inline README.md]:
```suggestion
VERSION="0.0.0-0.nightly.YYYY.MM.DD"
```

File: README.md [hunk 3]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
...
+
+# Unpack the toolchain:
+tar -xvf carbon_toolchain-${VERSION}.tar.gz
+
+# Create a simple Carbon source file:
+echo "fn Run() { Print(42); }" > forty_two.carbon
```
Comment (jonmeow) [inline README.md]:
```suggestion
echo "fn Run() { Core.Print(42); }" > forty_two.carbon
```

File: README.md [hunk 4]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
...
+# Link to an executable:
+./carbon_toolchain-${VERSION}/bin/carbon link \
+  --output=forty_two forty_two.o
+
+# Run it:
+./forty_two
+```
+
+However, the toolchain is still very early and many things don't yet work.
+Please hold off on filing lots of bugs, we know many parts of this don't work
```
Comment (jonmeow) [inline README.md]:
```suggestion
Please hold off on filing lots of bugs: we know many parts of this don't work
```

File: README.md [hunk 5]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
...
+yet or may not work on all systems. We expect to have releases that are much
+more robust and reliable that you can try out when we reach our
 [0.1 milestone](/docs/project/milestones.md#milestone-01-a-minimum-viable-product-mvp-for-evaluation).
 
-If you do want to try out Carbon locally, you'll need to install our
+If you do want to build Carbon's toolchain yourself and maybe contribute fixes
```
Comment (jonmeow) [inline README.md]: How much value do we get from providing both the release instruction and the `apt install` instruction here? Would it be worth removing the latter?

**Review by KateGregory — APPROVED** [review]
Happy to see this!

File: README.md [hunk 6]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
...
+
+# Create a simple Carbon source file:
+echo "fn Run() { Print(42); }" > forty_two.carbon
+
+# Compile to an object file:
+./carbon_toolchain-${VERSION}/bin/carbon compile \
+  --output=forty_two.o forty_two.carbon
+
+# Link to an executable:
+./carbon_toolchain-${VERSION}/bin/carbon link \
```
Comment (jonmeow) [inline README.md:304]: The latter instructions include clang, libc++, and lld. Do we need any of that in order to link?

**Review by jonmeow — COMMENTED** [review]
Looks good, but suggesting corrected instructions.

File: README.md [hunk 7]
```diff
@@ -271,13 +271,44 @@ To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
 print its output. You can try it out immediately at
 [compiler-explorer.com](http://carbon.compiler-explorer.com/).
 
-Because Carbon is an early, experimental project we don't yet have releases you
-can download and try out locally, you'll instead need to build any tools
-yourself from source. We expect to have packaged releases you can try out when
-we reach our
+Carbon is an early, experimental project, and so we only have very experimental
+nightly releases of the Carbon toolchain available to download, and only on
```
Comment (jonmeow) [inline README.md]: FWIW, this feels a bit odd coming after the "explorer" advice above. [I'd lean towards removing the explorer device, or moving it below]

File: README.md (hunk 6 above)
Comment (chandlerc) [reply README.md:304]: I don't *think* so. Definitely don't need clang or lld.

I'll try this out with a fresh Ubuntu cloud image just to see what if anything might be needed there.

@KateGregory -- anything we should include for WSL?

File: README.md (hunk 7 above)
Comment (chandlerc) [reply README.md]: I'd like to keep opening with the compiler-explorer link that folks can just dive into.

But I agree we need a transition in the text... I've tried to reword both sections a bit to emphasize that the above is just about trying stuff out directly in the browser. I've also removed the explorer reference from a local build below to focus in on the toolchain where we would potentially be interested in contributions.

PR description also updated a bit to reflect the changes here.

File: README.md (hunk 5 above)
Comment (chandlerc) [reply README.md]: I think it's nice to keep the source build snippets for now. I'm still interested in nudging folks interested to consider contributing.

I did prune the explorer part out as that's not really relevant for contributing at this point.

**Review by chandlerc — COMMENTED** [review]
Thanks! (Still need to try this out on a fresh ubuntu install though)

**Comment by eyelash** [comment]
Further up in the README it says:

```
Carbon Language is currently an experimental project. There is no working compiler or toolchain.
```

This sounds a bit contradictory to me with the instructions on how to download the toolchain you are adding here.

Also are there any plans to update compiler-explorer.com to use the full toolchain instead of the explorer? According to [p3532](https://github.com/chandlerc/carbon-lang/blob/trunk/proposals/p3532.md) the focus of the project has now shifted away from the explorer.

File: README.md (hunk 7 above)
Comment (jonmeow) [reply README.md]: NB, I think you might've forgotten to push the commit for this.

Comment (chandlerc) [reply README.md]: I... think it is pushed? Maybe it wasn't as much of an adjustment as you were looking for?

**Comment by chandlerc** [comment]
> Further up in the README it says:
> 
> ```
> Carbon Language is currently an experimental project. There is no working compiler or toolchain.
> ```
> 
> This sounds a bit contradictory to me with the instructions on how to download the toolchain you are adding here.

Updated that text to reflect the current progress on the toolchain.

> 
> Also are there any plans to update compiler-explorer.com to use the full toolchain instead of the explorer? According to [p3532](https://github.com/chandlerc/carbon-lang/blob/trunk/proposals/p3532.md?rgh-link-date=2024-07-24T15%3A30%3A45Z) the focus of the project has now shifted away from the explorer.

We'd like the toolchain to catch up a bit more to the amount of Carbon's design implemented before we switch over, but its definitely something we're working on.

File: README.md [hunk 8]
```diff
@@ -267,14 +268,15 @@ semantics onto C++ such as Rust-inspired
 
 ## Getting started
 
-To try out Carbon, you can use the Carbon explorer to interpret Carbon code and
-print its output. You can try it out immediately at
+To try out Carbon immediately in your browser, you can use the Carbon explorer
+to interpret Carbon code and print its output on:
```
Comment (jonmeow) [inline README.md]: You might consider putting a little more emphasis on "demo interpreter for Carbon", like above, rather than just "Carbon".

**Review by jonmeow — APPROVED** [review]
LGTM, I see the commit now.

File: README.md (hunk 6 above)
Comment (chandlerc) [reply README.md:304]: I've added the minimal necessary `sudo apt install` based on my testing right before linking (it isn't needed until linking).

File: README.md (hunk 8 above)
Comment (chandlerc) [reply README.md]: Done I think, PTAL.

**Review by chandlerc — COMMENTED** [review]
Tested on a fresh Ubuntu 22.04 LTS install, so I think this is good-to-go, but I did make some changes so PTAL.

**Review by chandlerc — COMMENTED** [review]
Tested on a fresh Ubuntu 22.04 LTS install, so I think this is good-to-go, but I did make some changes so PTAL.

File: README.md [hunk 9]
```diff
@@ -295,6 +295,11 @@ echo "fn Run() { Core.Print(42); }" > forty_two.carbon
 ./carbon_toolchain-${VERSION}/bin/carbon compile \
   --output=forty_two.o forty_two.carbon
 
+# Install minimal system libraries used for linking, note that installing `gcc`
```
Comment (jonmeow) [inline README.md]:
```suggestion
# Install minimal system libraries used for linking. Note that installing `gcc`
```

File: README.md [hunk 10]
```diff
@@ -295,6 +295,11 @@ echo "fn Run() { Core.Print(42); }" > forty_two.carbon
...
+# or `g++` for compiling C/C++ code with GCC will also be sufficient, this is
+# just the specific system libraries Carbon linking currently still uses.
```
Comment (jonmeow) [inline README.md]:
```suggestion
# or `g++` for compiling C/C++ code with GCC will also be sufficient, this is
# just the specific system library Carbon linking currently uses.
```

```suggestion
# or `g++` for compiling C/C++ code with GCC will also be sufficient, these are
# just the specific system libraries Carbon linking still uses.
```

**Review by jonmeow — APPROVED** [review]

**Comment by jonmeow** [comment]
👍

**Review by Bonniemarie216 — APPROVED** [review]

**Pull request merged**